artifactory/commands/transferinstall/installplugin_test.go

```diff
@@ -0,0 +1,261 @@
+package transferinstall
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/utils"
+	commonTests "github.com/jfrog/jfrog-cli-core/v2/common/tests"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/tests"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+func TestPluginFileItemNameAndDirs(t *testing.T) {
+	testFileItemNameDirs(t, PluginFileItem{"dir", "dir2"}, "name.txt", PluginFileItem{"dir", "dir2", "name.txt"})
+	testFileItemNameDirs(t, PluginFileItem{}, "name.txt", PluginFileItem{"name.txt"})
+	testFileItemNameDirs(t, PluginFileItem{}, "", PluginFileItem{})
+	testFileItemNameDirs(t, PluginFileItem{}, "", PluginFileItem{""})
+	testFileItemNameDirs(t, PluginFileItem{}, "", PluginFileItem{"", "", ""})
+	testFileItemNameDirs(t, PluginFileItem{"dir"}, "", PluginFileItem{"", "dir", ""})
+	testFileItemNameDirs(t, PluginFileItem{"dir"}, "name", PluginFileItem{"dir", "", "name"})
```

Comment:
To remove code duplication in tests, please use table testing. For example:
```suggestion
	cases := []struct {
		expectedDirs PluginFileItem
		expectedName string
		file         PluginFileItem
	}{
		{PluginFileItem{"dir", "dir2"}, "name.txt", PluginFileItem{"dir", "dir2", "name.txt"}},
		{PluginFileItem{}, "name.txt", PluginFileItem{"name.txt"}},
	}

	for _, testCase := range cases {
		t.Run("", func(t *testing.T) {
			testFileItemNameDirs(t, testCase.expectedDirs, testCase.expectedName, testCase.file)
		})
	}
```
Same goes for `TestPluginFileItemToUrlAndToPath`.

Reply:
done